Yoshi/Yoshi/YoshiActionResult.swift

```diff
@@ -1,42 +1,53 @@
 //
-//  YoshiActionResult.swift
+//  YoshiResult.swift
 //  Yoshi
 //
-//  Created by Christopher Jones on 2/8/16.
+//  Created by Christopher Jones on 2/9/16.
 //  Copyright © 2016 Prolific Interactive. All rights reserved.
 //
 
 /**
- A debug menu option result
- */
-public struct YoshiActionResult {
+The result types for a debug menu action.
 
-    /// The result of the action.
-    let result: YoshiResult
-    /// A view controller; this will be nil in the event that the result is .Handled.
-    let viewController: UIViewController?
+- Handled:               Indicates that the action was handled and nothing else is required.
+- PresentViewController: Indicates that a view is required to complete the action, returning the view controller to present.
+- AsyncAfterDismissing:  Indicates that the action should be handled asynchronously after the view is dismissed.
+*/
+public enum YoshiActionResult: Equatable {
 
-    /**
-     An empty result, indicating that the action was handled and no further work is necessary.
-     */
-    public init() {
-        result = .Handled
-        viewController = nil
+    case Handled
+    case PresentViewController(UIViewController)
+    case AsyncAfterDismissing(() -> Void)
+
+    func asyncBlock() -> () -> Void {
+        switch self {
+        case .AsyncAfterDismissing(let block):
+            return block
+        default:
+            return { }
+        }
     }
 
-    public init(action: () -> Void) {
-        result = .AsyncAfterDismissing(action)
-        viewController = nil
+    func viewController() -> UIViewController? {
```

Comment:
Do we really need these two functions and the `Equatable` implementation here? Isn't the Swift built-in pattern matching enough for our purpose? E.g.

```
        switch result {
        case .PresentViewController(let viewController):
            // do things to view controller
        case .AsyncAfterDismissing(let asyncBlock):
            // call or pass block
        default:
            return
        }
```